### PR TITLE
Provide the charm store API instance to remaining addCharm calls.

### PR DESCRIPTION
**Pull request by frankban**
So that it is possible to upgrade/downgrade charms for existing applications, and also deploy bundles with non-public charms or, charms with terms.

### CODE REVIEW

**Comment by frankban** [comment]
QA:
- deploy the gitlab ssl bundle; which includes a charm with terms;
- when done, change the version of the ssl-termination-proxy charm.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7261/

**Comment by hatched** [comment]
The current bakery implementation doesn't send the correct headers for the terms request. So QA is on hold until the fix has landed.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7286/

**Comment by frankban** [comment]
I'll have to fix some tests but, as I rebased to current new-bakery, it should be possible to QA this now.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7288/

**Comment by frankban** [comment]
Ok now the tests pass.

File: jujugui/static/gui/src/app/components/inspector/change-version/test-change-version.js [hunk 1]
```diff
@@ -309,7 +308,7 @@ describe('InspectorChangeVersion', function() {
       get: sinon.stub().returns('django'),
       set: serviceSet
     };
-    var addCharm = sinon.stub().callsArgWith(2, {err: 'error'});
+    var addCharm = sinon.stub().callsArgWith(1, {err: 'error'});
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/inspector/change-version/test-change-version.js]: consts/ lets

Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector/change-version/test-change-version.js]: Done.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7291/

**Comment by Lukewh** [comment]
QA OK @frankban

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**